src/Components/PropertyInspector/PropertyInspectoryModel.ts

```diff
@@ -344,21 +384,116 @@ abstract class PropertyInspectorModel {
+                                                  modelProperty.name,
+                                                  modelProperty.index
                                               ),
                                               propertySourceObject: mapValue,
                                               mapInfo: { key },
                                               isMapChild: true,
                                               forceSet: true,
-                                              schemas
+                                              schemas,
+                                              isArrayItem
                                           }
                                       );
                                   })
                                 : null,
                         ...(schemas && { mapSchemas: schemas })
                     };
                 }
-                case DTDLSchemaType.Array: // TODO support arrays in future
+                case DTDLSchemaType.Array: {
+                    const arrayItems: any[] = PropertyInspectorModel.getPropertyValueOrDefault(
+                        modelProperty.name,
+                        propertySourceObject,
+                        dtdlPropertyTypesEnum.Array
+                    );
+                    const targetSchema = schemas?.find(
+                        (s) =>
+                            s['@id'] ===
+                            ((modelProperty.schema as any).elementSchema ||
+                                modelProperty.schema)
+                    );
+                    return {
+                        name: modelProperty.name,
+                        displayName: PropertyInspectorModel.parsePropertyTreeDisplayName(
+                            {
+                                name: modelProperty.name,
+                                ...modelProperty
+                            }
+                        ),
+                        role: NodeRole.parent,
+                        schema: dtdlPropertyTypesEnum.Array,
+                        isCollapsed: true,
+                        type: DTDLType.Property,
+                        path: mapInfo
+                            ? PropertyInspectorModel.buildPath(
+                                  path,
+                                  mapInfo.key
+                              )
+                            : PropertyInspectorModel.buildPath(
+                                  path,
+                                  modelProperty.name,
+                                  modelProperty.index
+                              ),
+                        parentObjectPath: isObjectChild && path,
+                        isInherited,
+                        isMapChild,
+                        isArrayItem,
+                        childSchema: modelProperty.schema.elementSchema,
+                        isRemovable: !(isMapChild || isArrayItem),
```

Comment:
don't we know that this isn't a map now that we are inside teh Array case condition?

Reply:
Good point. This'll always be falsey. Will remove.